redaxo/src/core/lib/rex.php

```diff
@@ -156,6 +156,30 @@ public static function isBackend()
     }
 
     /**
+     * Returns if the environment is the console.
+     *
+     * @return bool
+     */
+    public static function isConsole()
+    {
+        return (bool) self::getProperty('console', false);
+    }
+
+    /**
+     * Returns the environment.
+     *
+     * @return string
+     */
+    public static function getEnvironment()
+    {
+        if (self::isConsole()) {
+            return 'console';
+        }
+
+        return self::isBackend() ? 'backend' : 'frontend';
```

Comment:
Hier bin ich mir noch nicht sicher mit den Methoden.
Aktuell ist es so:
`isBackend()` liefert `true` sowohl bei normalen Backend-Aufrufen, als auch bei der console-Applikation. Diese wird also auch im Backend-Kontext ausgeführt. Ich denke, dass das sinnvoll ist.
`getEnvironment` hingegen liefert aber `console`  in der Konsole-Applikation, und `backend` nur wirklich bei normalen Backend-Aufrufen. Die Methode habe ich insbesondere für diese virtuellen Usernamen (siehe Diff) hinzugefügt.

Ist das so zu verwirrend?